### ISSUE TITLE
sg_get_process_stats_int() leaving open file when process name have "spaces" on it

### ISSUE THREAD

**Issue by andersontrindade0975** [i-scream/libstatgrab#76]
Hi,

On a Linux Server, I have a process running which has spaces on his name as below:

$ps -ef | grep Mtx
...
root      2916     2  0 Apr01 ?        00:00:00 [MpxTestDaemon  ]

When the function  sg_get_process_stats_int() is being called (indirectly) It's seems this function is leaving open file descriptors for the "stat" file of the process above, as you can see:

$lsof -p 29785 (<- this PID is of my process which uses libstatgrab)

sprof   29785 optimode  cwd    DIR 253,18     4096          2 /tmp
sprof   29785 optimode  rtd    DIR  253,0     4096          2 /
sprof   29785 optimode  txt    REG  253,0   613441     291040 /opt/optimode/bin/sprof
...
sprof   29785 optimode  340r   REG    0,3        0      51408 /proc/2916/stat
sprof   29785 optimode  341r   REG    0,3        0      51408 /proc/2916/stat
sprof   29785 optimode  342r   REG    0,3        0      51408 /proc/2916/stat
sprof   29785 optimode  343r   REG    0,3        0      51408 /proc/2916/stat
sprof   29785 optimode  344r   REG    0,3        0      51408 /proc/2916/stat
sprof   29785 optimode  345r   REG    0,3        0      51408 /proc/2916/stat
...
sprof   29785 optimode  825r   REG    0,3        0      51408 /proc/2916/stat
sprof   29785 optimode  826r   REG    0,3        0      51408 /proc/2916/stat
sprof   29785 optimode  827r   REG    0,3        0      51408 /proc/2916/stat
sprof   29785 optimode  828r   REG    0,3        0      51408 /proc/2916/stat

I could workaround this problem adding a fclose()  on line 473 of process_stats.c, as below:

```
470                 if( ( f = fopen(filename, "r") ) == NULL ) {
471                         /* Open failed.. Process since vanished, or the path was too long.
472                          * Ah well, move onwards to the next one */
473                         fclose(f);
474                         ++pid_item;
475                         continue;
476                 }
```

**Comment by andersontrindade0975** [comment]
Just one correction, the workaround is a fclose() in the line 485 and not in the line 473 as stated before:

    479                 if( fscanf(f, FMT_PID_T "%4095s %c " FMT_PID_T " " FMT_PID_T " %*d %*d %*d %*u %*u %*u %*u %*u %lu %lu %*d %*d %*d %d %*d %*d %llu %llu %llu %*u %*u %*u %*u %*u %*u %*u %*u %*        u %*u %*u %*u %*u %*d %*d\n",
    480                            &proc_stats_ptr[proc_items].pid, read_buf, &s, &proc_stats_ptr[proc_items].parent,
    481                            &proc_stats_ptr[proc_items].pgid, &utime, &stime, &proc_stats_ptr[proc_items].nice,
    482                            &starttime, &proc_stats_ptr[proc_items].proc_size, &proc_stats_ptr[proc_items].proc_resident) < 11 ) {
    483                         /* Read failed.. Process vanished?
    484                          * Ah well, move onwards to the next one */
    485                         fclose(f);
    486                         ++pid_item;
    487                         continue;

**Comment by tdb** [comment]
Thanks for reporting this. Looks like it's already fixed in the master branch 84fd8c38ee4feb0117ed22ab56f6b46661411ec6? So it'll be in the next release. It'd be great if you could grab the latest copy from Git and give it a whirl.

**Comment by rehsack** [comment]
The root cause implies two issues - the second one is fixed in commit https://github.com/i-scream/libstatgrab/commit/ca9fa7749d7773910e9870cafcf4862f0ff1b39a :-)
